### ISSUE TITLE
ModuleNotFoundError when starting app after installing production dependencies

### ISSUE THREAD

**Issue by VKTB** [ral-facilities/datagateway-api#358]
**Description**:
Trying to start the app after installing the production dependencies using `poetry install --no-dev` throws this error:
```
datgateway_api_container | Traceback (most recent call last):
datgateway_api_container |   File "/usr/local/bin/poetry", line 5, in <module>
datgateway_api_container |     from poetry.console import main
datgateway_api_container |   File "/usr/local/lib/python3.6/site-packages/poetry/console/__init__.py", line 1, in <module>
datgateway_api_container |     from .application import Application
datgateway_api_container |   File "/usr/local/lib/python3.6/site-packages/poetry/console/application.py", line 7, in <module>
datgateway_api_container |     from .commands.about import AboutCommand
datgateway_api_container |   File "/usr/local/lib/python3.6/site-packages/poetry/console/commands/__init__.py", line 2, in <module>
datgateway_api_container |     from .add import AddCommand
datgateway_api_container |   File "/usr/local/lib/python3.6/site-packages/poetry/console/commands/add.py", line 8, in <module>
datgateway_api_container |     from .init import InitCommand
datgateway_api_container |   File "/usr/local/lib/python3.6/site-packages/poetry/console/commands/init.py", line 14, in <module>
datgateway_api_container |     from tomlkit import inline_table
datgateway_api_container | ModuleNotFoundError: No module named 'tomlkit'
```

**Acceptance criteria**:
- [ ] App should start after installing production dependencies

**Comment by MRichards99** [comment]
Oh yeah good point, I forgot about that

**Comment by VKTB** [comment]
Looks like `tomlkit` is used by the `python-semantic-release` package
https://github.com/ral-facilities/datagateway-api/blob/c67420548dd4cf951c31195362557496880ce7bb/poetry.lock#L905

This package is listed as a dev dependency
https://github.com/ral-facilities/datagateway-api/blob/c67420548dd4cf951c31195362557496880ce7bb/pyproject.toml#L58

Making it a non-dev dependency fixes the problem but I would have thought that the semantic release is only needed for development. @MRichards99 Any ideas why Poetry requires this to be installed when running `poetry install --no-dev`?

**Comment by MRichards99** [comment]
It looks like a dependency that Poetry has, probably so Poetry can deal with the `pyproject.toml` files: https://github.com/python-poetry/poetry/blob/master/pyproject.toml#L56

I don't see adding it as a non-dev dependency is a problem though. I wonder if this bug started because of the `python-semantic-release` library - perhaps when we added it as a dev dependency, Poetry has got a little confused and thinks `tomlkit` is only a dev dependency when it actually needs to be a prod dependency.

**Comment by VKTB** [comment]
I think it has also confused other dependencies from that list because when I manually installed `tomlkit` during my investigation, it then threw a `html5lib` error.

It was just that the `python-semantic-release` library is not needed for production and installing it takes up an extra 36MB. From a Docker perspective, image size matters a lot so I thought I would check if you had an idea.

**Comment by MRichards99** [comment]
`html5lib` isn't in the Poetry lock file so I'm not sure what's going on there. Maybe you could install `tomlkit` at the same time as installing Poetry? So for the CI, we pip install Poetry, so you could also pip install `tomlkit` too and see if that fixes the issue (I'm not sure if it will, but worth a try)

**Comment by VKTB** [comment]
`pip install tomlkit` didn't work which is why I had to manually install it in the container. Later I discovered that adding it to the `pyproject.toml` file took care of that. However, in both cases I was getting the below error which from the file paths, I think is coming from the `html5lib` library(?)
```
datgateway_api_container | Traceback (most recent call last):
datgateway_api_container |   File "/usr/local/bin/poetry", line 5, in <module>
datgateway_api_container |     from poetry.console import main
datgateway_api_container |   File "/usr/local/lib/python3.6/site-packages/poetry/console/__init__.py", line 1, in <module>
datgateway_api_container |     from .application import Application
datgateway_api_container |   File "/usr/local/lib/python3.6/site-packages/poetry/console/application.py", line 7, in <module>
datgateway_api_container |     from .commands.about import AboutCommand
datgateway_api_container |   File "/usr/local/lib/python3.6/site-packages/poetry/console/commands/__init__.py", line 4, in <module>
datgateway_api_container |     from .check import CheckCommand
datgateway_api_container |   File "/usr/local/lib/python3.6/site-packages/poetry/console/commands/check.py", line 2, in <module>
datgateway_api_container |     from poetry.factory import Factory
datgateway_api_container |   File "/usr/local/lib/python3.6/site-packages/poetry/factory.py", line 18, in <module>
datgateway_api_container |     from .repositories.pypi_repository import PyPiRepository
datgateway_api_container |   File "/usr/local/lib/python3.6/site-packages/poetry/repositories/pypi_repository.py", line 15, in <module>
datgateway_api_container |     from html5lib.html5parser import parse
datgateway_api_container |   File "/usr/local/lib/python3.6/site-packages/html5lib/__init__.py", line 25, in <module>
datgateway_api_container |     from .html5parser import HTMLParser, parse, parseFragment
datgateway_api_container |   File "/usr/local/lib/python3.6/site-packages/html5lib/html5parser.py", line 6, in <module>
datgateway_api_container |     from . import _inputstream
datgateway_api_container |   File "/usr/local/lib/python3.6/site-packages/html5lib/_inputstream.py", line 10, in <module>
datgateway_api_container |     import webencodings
datgateway_api_container | ModuleNotFoundError: No module named 'webencodings'
```

**Comment by VKTB** [comment]
I discovered that these errors are only thrown when using the `poetry run` command to run the app - `poetry run python -m datagateway_api.src.main` or `poetry run gunicorn -c gunicorn.conf.py datagateway_api.wsgi:application`.

No errors when using `python -m datagateway_api.src.main` or `gunicorn -c gunicorn.conf.py datagateway_api.wsgi:application`.

**Comment by VKTB** [comment]
After speaking with one of the main Poetry contributors on Discord, I finally understand why this is happening.

Basically this error was happening in the DG API Docker container that I have been working on and in its `Dockerfile` I have a command that disables the `venv` - `poetry config virtualenvs.create false`. Turns out I should not be doing that and removing that command fixes this problem.

I thought (and many other people online have the same thoughts) that a `venv` is not needed in a container which is why I disabled it to start with. Ther person did say that we won't encounter this problem in Poetry 1.2 as it will not auto-remove packages that are not in the tree.
